### ISSUE TITLE
Investigate which int overloads are still needed (exp, pow, ...)

### ISSUE THREAD

**Issue by rok-cesnovar** [stan-dev/math#1692]
## Description

My lab mate asked me about the reasoning behind having `exp(int)` defined in the Math library. 

Example: `inline double exp(int x) { return std::exp(x); }` in https://github.com/stan-dev/math/blob/develop/stan/math/prim/fun/exp.hpp#L18

I could not find a reason so I am wondering myself also. Can anyone help me understand the reasoning behind this.

This came up while trying to add a custom function in stan math and using something like `exp(3.3)` inside the stan::math namespace was being truncated down to `exp(3)`. This was due to a missing `using std::exp;` that would have been picked if he used cpplint.

#### Current Version:
v3.1.0

**Comment by bob-carpenter** [comment]
That problem hadn't occurred to me!

The reason we have the `int` signature is to avoid ambiguity with `exp(int)` when `std::exp()` and `stan::math::exp()` are in scope.  None of them have integer signatures, so an `int` either needs to be promoted to `double` or to `var` or `fvar`. 

We could get around this by having our math lib version be defined as something like:

```
template <typename T, typename = std::enable_if_t<std::is_arithmetic<scalar_type_t<T>>::value>>
inline T exp(T x) {
  return std::exp(x);
}
```

Then it won't try to cast `double` to `int`, while still being a better match for `exp(int)` than `std::exp()`.

**Comment by bob-carpenter** [comment]
OK, I just went and looked up the [signatures of `exp`](https://en.cppreference.com/w/cpp/numeric/math/exp) and as of C++11, there's an `exp(IntegralType)`.  That must not cause ambiguities with our `exp(int)`, but it probably means we can get rid of our `exp(int)` definition without problems.

**Comment by rok-cesnovar** [comment]
> while still being a better match for `exp(int)` than `std::exp()`

But stan::math::exp still uses std::exp so these would be the same in the end right? So I guess this was to prevent int being promoted to var?

**Comment by bob-carpenter** [comment]
Right---the exp(int) signature was just to disambiguate calls to exp(int) in Stan, which were otherwise ambiguous betwene promotion to double and promotion to var.

**Comment by rok-cesnovar** [comment]
This was fixed in #1711
